pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/ast/SafeNavigationOperator.txt

```diff
@@ -1,99 +1,91 @@
-      +- ModifierNode[@Abstract = false, @DefiningType = "Foo", @DeprecatedTestMethod = false, @Final = false, @Global = false, @InheritedSharing = false, @Modifiers = 1, @Namespace = "", @Override = false, @Private = false, @Protected = false, @Public = true, @RealLoc = true, @Static = false, @Test = false, @TestOrTestSetup = false, @Transient = false, @Virtual = false, @WebService = false, @WithSharing = false, @WithoutSharing = false]
-      +- Field[@DefiningType = "Foo", @Image = "x", @Name = "x", @Namespace = "", @RealLoc = true, @Type = "Integer", @Value = null]
-      |  +- ModifierNode[@Abstract = false, @DefiningType = "Foo", @DeprecatedTestMethod = false, @Final = false, @Global = false, @InheritedSharing = false, @Modifiers = 0, @Namespace = "", @Override = false, @Private = false, @Protected = false, @Public = false, @RealLoc = true, @Static = false, @Test = false, @TestOrTestSetup = false, @Transient = false, @Virtual = false, @WebService = false, @WithSharing = false, @WithoutSharing = false]
-      +- Field[@DefiningType = "Foo", @Image = "profileUrl", @Name = "profileUrl", @Namespace = "", @RealLoc = true, @Type = "String", @Value = null]
-      |  +- ModifierNode[@Abstract = false, @DefiningType = "Foo", @DeprecatedTestMethod = false, @Final = false, @Global = false, @InheritedSharing = false, @Modifiers = 0, @Namespace = "", @Override = false, @Private = false, @Protected = false, @Public = false, @RealLoc = true, @Static = false, @Test = false, @TestOrTestSetup = false, @Transient = false, @Virtual = false, @WebService = false, @WithSharing = false, @WithoutSharing = false]
-      +- FieldDeclarationStatements[@DefiningType = "Foo", @Namespace = "", @RealLoc = true, @TypeName = "Integer"]
-      |  +- ModifierNode[@Abstract = false, @DefiningType = "Foo", @DeprecatedTestMethod = false, @Final = false, @Global = false, @InheritedSharing = false, @Modifiers = 0, @Namespace = "", @Override = false, @Private = false, @Protected = false, @Public = false, @RealLoc = false, @Static = false, @Test = false, @TestOrTestSetup = false, @Transient = false, @Virtual = false, @WebService = false, @WithSharing = false, @WithoutSharing = false]
-      |  +- FieldDeclaration[@DefiningType = "Foo", @Image = "anIntegerField", @Name = "anIntegerField", @Namespace = "", @RealLoc = true]
-      |     +- VariableExpression[@DefiningType = "Foo", @Image = "anIntegerField", @Namespace = "", @RealLoc = true]
-      |     |  +- ReferenceExpression[@Context = null, @DefiningType = "Foo", @Namespace = "", @RealLoc = false, @ReferenceType = ReferenceType.LOAD, @SObjectType = false, @SafeNav = true]
-      |     |     +- VariableExpression[@DefiningType = "Foo", @Image = "anObject", @Namespace = "", @RealLoc = true]
++- ApexFile[@DefiningType = null, @Namespace = null, @RealLoc = true]
+   +- UserClass[@DefiningType = "c__Foo", @Image = "Foo", @Namespace = null, @RealLoc = true, @SimpleName = "Foo", @SuperClassName = ""]
```

Comment:
The defining type is now `c__Foo` rather than just `Foo`. Is this correct?

Reply:
It turns out, that ApexQualifiedName used "c" as the namespace, if the namespace was not known. So, the fully qualified name included this.
However, the namespace was never known in PMD. It was always an empty string when using Jorje. When parsing, we could have defined the namespace here:

https://github.com/pmd/pmd/blob/19496ef198ea79ec7ff516e44744e2ff8887076f/pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/CompilerService.java#L64-L67

There we could have called `setNamespace(Namespaces.create("mynamespace", "mymodule"))` and we also could have called `setVersion` to set the version.

However, both information are defined outside of each individual file, e.g. in a sfdx-project.json, see https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_config.htm

I decided to remove this "c__" prefix and remove the namespace implementation entirely.